### PR TITLE
Make beaker-docker work in a docker container

### PR DESCRIPTION
**Pull request by Ananasr**
Hey ! Please review and let me know what you think!

### CODE REVIEW

**Comment by rishijavia** [comment]
@Ananasr, we follow [beaker's contributing guideline](https://github.com/puppetlabs/beaker/blob/master/CONTRIBUTING.md) for all our hypervisor libraries.

Would you mind filing a ticket and also describing the problem in detail for us to review this patch?

**Comment by rishijavia** [comment]
Also what do you think about fixing the port mapping in the first place itself? I made this [change](https://github.com/rishijavia/beaker-docker/commit/066e72d839a67f85f7e6ec38b9bdfa824da93d28) on my fork.

Docker changes the port mapping for port 22 every time the container restarts. It would be easier for us to just hardcode this mapping to a random port during the creation of the container so that it would not change on restart and it'd be easy for us to SSH.

**Comment by Ananasr** [comment]
Can you show me a use case where the current port mapping is a problem ?

**Comment by rishijavia** [comment]
This [test](https://github.com/puppetlabs/beaker/blob/master/acceptance/tests/base/host/reboot_test.rb) will fail with how we map port currently.

**Comment by Ananasr** [comment]
This fix the problem of rebooting ! Cool !

**Comment by rishijavia** [comment]
@Ananasr I still don't understand the problem you were trying to solve. Does the patch i proposed fixes it? In that case I can create a ticket for my patch and submit a PR for that.

**Comment by Ananasr** [comment]
If I try to use `beaker-docker` inside a container for testing using `docker run -it -v /var/run/docker.sock:/var/run/docker.sock ruby bash` (for example). Here, the container is able to connect to the Docker daemon, but `beaker` is trying to connect at `0.0.0.0` which is the container, not the host ... Am I clear ?

**Comment by rishijavia** [comment]
Ahh so there are two problems here:
1. The change of port mapping on reboot i mentioned
2. `beaker-docker` not being able to run from within a container

Right? If so, we should file a ticket for both and submit PRs following the contributing guidelines for them to be verified.

**Comment by Ananasr** [comment]
Exactly ! I'll file a ticket now !

**Comment by Ananasr** [comment]
Here is my ticket https://tickets.puppetlabs.com/browse/BKR-1186

**Comment by rishijavia** [comment]
@Ananasr couple more steps to go:
1. Update your PR to match the [contributing guideline](https://github.com/puppetlabs/beaker/blob/master/CONTRIBUTING.md#making-changes). Update PR and commit with ticket number.
2. Add spec tests to it

**Comment by Ananasr** [comment]
I'v some questions about writing spec tests, where can I find help ?

**Comment by rishijavia** [comment]
@Ananasr this looks good! Kicking off CI

**Comment by rishijavia** [comment]
@puppetlabs-jenkins test this please

File: lib/beaker/hypervisor/docker.rb [hunk 1]
```diff
@@ -332,5 +336,10 @@ def find_container(host)
       end.first
     end
 
+    # return true if we are inside a docker container
+    def am_i_in_container
```
Comment (rishijavia) [inline lib/beaker/hypervisor/docker.rb]: in ruby we use the convention of using `?` if it's a method that returns a boolean. So if you were to change this method name to `in_container?`, I think everything else looks great!

sorry for being a but nit picky about this.

Comment (Ananasr) [inline lib/beaker/hypervisor/docker.rb]: No problem @rishijavia ! I'll fix this right now !

**Comment by rishijavia** [comment]
@puppetlabs-jenkins retest this please

**Pull request merged**